cloudinit/config/cc_ca_certs.py

```diff
@@ -44,60 +44,109 @@
 from cloudinit import subp
 from cloudinit import util
 
-CA_CERT_PATH = "/usr/share/ca-certificates/"
-CA_CERT_FILENAME = "cloud-init-ca-certs.crt"
-CA_CERT_CONFIG = "/etc/ca-certificates.conf"
-CA_CERT_SYSTEM_PATH = "/etc/ssl/certs/"
-CA_CERT_FULL_PATH = os.path.join(CA_CERT_PATH, CA_CERT_FILENAME)
+DEFAULT_CONFIG = {
+    'ca_cert_path': '/usr/share/ca-certificates/',
+    'ca_cert_filename': 'cloud-init-ca-certs.crt',
+    'ca_cert_config': '/etc/ca-certificates.conf',
+    'ca_cert_system_path': '/etc/ssl/certs/',
+    'ca_cert_update_cmd': ['update-ca-certificates']
+}
+DISTRO_OVERRIDES = {
+    'rhel': {
+        'ca_cert_path': '/usr/share/pki/ca-trust-source/',
+        'ca_cert_filename': 'anchors/cloud-init-ca-certs.crt',
+        'ca_cert_config': None,
+        'ca_cert_system_path': '/etc/pki/ca-trust/',
+        'ca_cert_update_cmd': ['update-ca-trust']
+    }
+}
 
-distros = ['alpine', 'debian', 'ubuntu']
 
+distros = ['alpine', 'debian', 'ubuntu', 'rhel']
 
-def update_ca_certs():
+
+def _distro_ca_certs_configs(distro):
+    """Return a distro-specific ca_certs config dictionary
+
+    @param distro: String providing the distro class name.
+    @returns: Dict of distro configurations for ca-cert.
+    """
+    cfg = DEFAULT_CONFIG
+    dcfg = DISTRO_OVERRIDES
+    if distro in dcfg:
+        cfg = dcfg[distro]
+
+    cfg['ca_cert_full_path'] = os.path.join(cfg['ca_cert_path'],
+                                            cfg['ca_cert_filename'])
+    return cfg
+
+
+def update_ca_certs(distro_name):
     """
     Updates the CA certificate cache on the current machine.
+
+    @param distro: String providing the distro class name.
     """
-    subp.subp(["update-ca-certificates"], capture=False)
+    distro_cfg = _distro_ca_certs_configs(distro_name)
```

Comment:
We call `_distro_ca_certs_config(distro_name)` in many different places in this PR. Do you think it would make sense to instead call it once (in `handle`) and then pass the configuration dict itself around?

Reply:
Okay, to update this, I have to update test case, too.
So I will try

Reply:
I updated, please confirm